pkg/envoy/lds/egress.go

```diff
@@ -32,15 +32,15 @@ func (lb *listenerBuilder) getEgressFilterChainsForMatches(matches []*trafficpol
 		switch match.DestinationProtocol {
 		case constants.ProtocolHTTP:
 			// HTTP protocol --> HTTPConnectionManager filter
-			if filterChain, err := lb.getEgressHTTPFilterChain(*match); err != nil {
+			if filterChain, err := lb.buildEgressHTTPFilterChain(*match); err != nil {
```

Comment:
*This comment is less of a code review, and more meant to elicit conversation.*

I think it's ever-so-slightly preferable to not pass fields that are on a struct to methods on that same struct, as it becomes easier in code navigation to see instances of that field in one place vs needing to click through to the next func.

That's mostly a pre-amble for the following comment:

I'm wondering, if while you've done this refactoring, if you've noticed instances where it might make sense to keep the data on the builder in a different structure. ie, maybe for match we have on the builder a `matchesByName map[string]*trafficpolicy.TrafficMatch`

It may also leave us inclined to refactor some of the meshcatalog methods, or split up some of the trafficpolicy objects.

Reply:
For instance in `buildEgressHTTPFilterChain`, it looks like only the Name and DestinationPort are used.

Maybe that means we can add another meshcatalog query `GetPortsByMatchForService(...)` or something of the sort. Maybe not, and definitely doesn't need to be done in this PR, but might be worth leaving some comments throughout the code if you've noticed places

Reply:
Some of the existing utility builders operate on TrafficMatch objects that are not a part of the builder struct directly but that may be generated by the caller within the `build()` function. I think in this case, it makes sense to pass the param to the function as we don't have it readily available on the method receiver.